dtfs-central-api/src/v1/controllers/utilisation-report-service/post-payment.controller/index.ts

```diff
@@ -28,7 +35,9 @@ export const postPayment = async (req: PostPaymentRequest, res: Response) => {
     };
 
     await addPaymentToUtilisationReport(parseInt(reportId, 10), feeRecordIds, user, newPaymentDetails);
-    return res.sendStatus(HttpStatusCode.Ok);
+    const { status } = await FeeRecordRepo.findOneByOrFail({ id: feeRecordIds[0] });
```

Comment:
I feel a bit mixed about this being included here vs in the transaction within `addPaymentToUtilisationReport`:

If `findOneByOrFail` rejects we'll end up in the catch which will produce an error on the frontend, but the payment would've been added.

It feels like this would ideally be in the same transaction as adding the payment (at the end of the same transaction that's being executed within `addPaymentToUtilisationReport`), then we can return it from the call to that maybe?

Will mean some of the tests need shifting too - apologies

Reply:
Good shout! Have made the change 😄